pkg/store/proxy_merge.go

```diff
@@ -139,8 +144,24 @@ func chainSeriesAndRemIdenticalChunks(series []*storepb.SeriesResponse) *storepb
 	}
 
 	finalChunks := make([]storepb.AggrChunk, 0, len(chunkDedupMap))
-	for _, chk := range chunkDedupMap {
-		finalChunks = append(finalChunks, *chk)
+	for hash, chk := range chunkDedupMap {
+		if quorum {
+			// NB: this is specific to Databricks' setup where each time series is written to at least 2 out of 3 replicas.
+			// Each chunk should have 3 replicas in most cases, and 2 replicas in the worst acceptable cases.
+			// Quorum-based deduplication is used to pick the majority value among 3 replicas.
+			// If a chunck has only 2 identical replicas, there might be another chunk with corrupt data.
+			// We want to send those two identical replicas to the later quorum-based deduplication process to dominate any corrupt third replica.
+			if chunckCountMap[hash] >= 3 {
+				// Most of cases should hit this branch.
+				finalChunks = append(finalChunks, *chk)
+			} else {
+				for i := 0; i < chunckCountMap[hash]; i++ {
+					finalChunks = append(finalChunks, *chk)
+				}
```

Comment:
nice debugging, previously the merge_iter only see 1 dedup'ed series instead of 3 identical ones!